javascripts/2048.js

```diff
@@ -1,32 +1,400 @@
+// Constants -----------------------------------------------------------
+const MINSTARTINGTILE = 2;
+const MAXSTARTINGTILE = 4;
+const MINBOARDLOCALE = 0; // starting array index
+const MAXBOARDLOCALE = 3; // highest array index
+const BOARDSIZE = 4; // anything less than 4 is valid
+const WINNINGTILE = 2048;
+// Global Variables ------------------------------------------------------
+var board;
+var score;
+var actionOccurred;
+var alreadyWon;
+var gameOver;
+
 $(document).ready(function() {
-  console.log('ready!');
-  $('body').keydown(function(event){
-    var arrow_keys = [37, 38, 39, 40];
-    if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
-      event.preventDefault();
+  begin();
```

Comment:
JavaScript will allow you to call a function that's defined later in the current scope, but it's generally frowned upon because it's hard to understand the flow of the program. Consider moving this line to the bottom of the `ready` callback.